js/main.js

```diff
@@ -0,0 +1,504 @@
+let recordsRowTemplate = document.querySelector("#recordsRowTemplate");
+let gamePausedFlag = false;
+
+let board = document.querySelector("#board");
+let pausedElement = document.querySelector("#pausedElement");
+let newGameBtn = document.querySelector("#newGameBtn");
+let pauseBtn = document.querySelector("#pauseBtn");
+let movesCountField = document.querySelector("#movesCountField");
+let timeCountField = document.querySelector("#timeCountField");
+let movesRecordsField = document.querySelector("#movesRecords");
+let timeRecordsField = document.querySelector("#timeRecords");
+
+let tiles;												//масив всіх плиток
+let puzzleIsSolved;											//перевірка чи розв'язана головоломка
+
+addMoveTilesEvents();
+restoreCurrentTilesPosition();
+makeStartBoardAndOptions(savedPositionData);
+updateRecordsDataFromStorage();
+updateRecordsFields();
+
+totalRowsField.addEventListener("change", function(){
+	makeStartBoardAndOptions();
+	updateRecordsFields();
+}, false);
+
+totalColumnsField.addEventListener("change", function(){
+	makeStartBoardAndOptions();
+	updateRecordsFields();
+}, false);
+
+newGameBtn.addEventListener("click", function(){
+	resetStopwatch();
+	movesCountField.innerHTML = 0;
+	shuffle();
+}, false);
+
+pauseBtn.addEventListener("click", pauseResumeGame, false);
+
+function makeStartBoardAndOptions(savedPosition) {
+	board.innerHTML = "";
+	tiles = [];
+	movesCountField.innerHTML = 0;
+	totalRows = parseInt(totalRowsField.value);
+	totalColumns = parseInt(totalColumnsField.value);
+	tileSize = maxBoardSize / Math.max(totalRows, totalColumns) - 10;		
+	board.style.width = (tileSize + 10) * totalColumns + "px";	//задаємо ширину поля з урахуванням відстані між плитками
+	board.style.height = (tileSize + 10) * totalRows + "px";	//задаємо висоту поля з урахуванням відстані між плитками
+	tilesQuantity = totalRows * totalColumns;
+	
+	if (savedPosition){
+		savedPosition.tilesPos.forEach(tile => {
+			tiles.push(new Tile(tile.numberOfTile, tile.column, tile.row, tileSize, tilesQuantity, tileMoveTime));
+		});
+		stopwatchCounter = savedPosition.stopwatchCounter;
+		movesCountField.innerHTML = savedPosition.moovesCount;
+		emptyTileindex = tiles.findIndex(tile => tile.numberOfTile === tilesQuantity);
+		puzzleIsSolved = false;
+		addPointerCursor();
+		pauseResumeGame();
+	} else {
+		puzzleIsSolved = true;
+		resetStopwatch();
+
+		for (let row = 1; row <= totalRows; row++){
+			for (let column = 1; column <= totalColumns; column++){
+				let numberOfTile = (row - 1) * totalColumns + column;
+				tiles.push(new Tile(numberOfTile, column, row, tileSize, tilesQuantity, tileMoveTime));
+			}
+		}
+	}
+	
+	tiles.forEach(tile => {
+		if (tile.tileElement) {		
+			board.append(tile.tileElement);
+		}
+	});	
+}
+
+function addMoveTilesEvents(){
+	board.addEventListener("click", function(e){
+		if (!puzzleIsSolved && e.target.connectedTile){
```

Comment:
Looks like you have duplicated code 100-105 and 108-113. You should move it to saparate method if it's possible